core/models.py

```diff
@@ -290,8 +290,10 @@ class LocationDoesNotExistException(Exception):
 def get_location(location_slug):
     if location_slug:
         try:
+            # TODO This looks like a bug. Since it's using first this will
+            # raise an exception, it'll just return None.
             location = Location.objects.filter(slug=location_slug).first()
```

Comment:
good catch! in practice we will pretty much always have at least one location but this is still inelegant.